diff -Naurp kodi-16.1-c6f6e0b/xbmc/view/ViewStateSettings.cpp kodi-16.1-c6f6e0b-patch/xbmc/view/ViewStateSettings.cpp
--- kodi-16.1-c6f6e0b/xbmc/view/ViewStateSettings.cpp	2016-04-26 10:06:23.000000000 +0300
+++ kodi-16.1-c6f6e0b-patch/xbmc/view/ViewStateSettings.cpp	2016-06-30 11:21:28.000000000 +0300
@@ -124,11 +124,14 @@ bool CViewStateSettings::Load(const TiXm
   pElement = settings->FirstChild(XML_GENERAL);
   if (pElement != NULL)
   {
+    /*
+    // Force setting level to expert
     int settingLevel;
     if (XMLUtils::GetInt(pElement, XML_SETTINGLEVEL, settingLevel, (const int)SettingLevelBasic, (const int)SettingLevelExpert))
       m_settingLevel = (SettingLevel)settingLevel;
     else
-      m_settingLevel = SettingLevelStandard;
+    */
+      m_settingLevel = SettingLevelExpert;
 
     const TiXmlNode* pEventLogNode = pElement->FirstChild(XML_EVENTLOG);
     if (pEventLogNode != NULL)
