diff -Naurp kodi-16.0-beta1-9a5ddf5/addons/resource.language.en_gb/resources/strings.po kodi-16.0-beta1-9a5ddf5-patch/addons/resource.language.en_gb/resources/strings.po
--- kodi-16.0-beta1-9a5ddf5/addons/resource.language.en_gb/resources/strings.po	2015-11-02 11:54:41.000000000 +0200
+++ kodi-16.0-beta1-9a5ddf5-patch/addons/resource.language.en_gb/resources/strings.po	2015-11-15 19:17:39.000000000 +0200
@@ -5370,7 +5370,7 @@ msgstr ""
 
 #: xbmc/windows/GUIWindowSystemInfo.cpp
 msgctxt "#12395"
-msgid "Battery level"
+msgid "Linux version"
 msgstr ""
 
 #. Used on the button to toggle add-on update mode in the Add-on Manager.
@@ -5470,7 +5470,7 @@ msgstr ""
 #: skin.confluence
 #: skin.retouched
 msgctxt "#13012"
-msgid "Exit"
+msgid "Live TV (VDR)"
 msgstr ""
 
 #: skin.confluence
diff -Naurp kodi-16.0-beta1-9a5ddf5/xbmc/guiinfo/GUIInfoLabels.h kodi-16.0-beta1-9a5ddf5-patch/xbmc/guiinfo/GUIInfoLabels.h
--- kodi-16.0-beta1-9a5ddf5/xbmc/guiinfo/GUIInfoLabels.h	2015-11-02 11:54:40.000000000 +0200
+++ kodi-16.0-beta1-9a5ddf5-patch/xbmc/guiinfo/GUIInfoLabels.h	2015-11-15 19:25:13.000000000 +0200
@@ -365,7 +365,7 @@
 #define SYSTEM_HAS_ADDON            711
 #define SYSTEM_ADDON_TITLE          712
 #define SYSTEM_ADDON_ICON           713
-#define SYSTEM_BATTERY_LEVEL        714
+#define SYSTEM_LINUX_VER            714
 #define SYSTEM_IDLE_TIME            715
 #define SYSTEM_FRIENDLY_NAME        716
 #define SYSTEM_SCREENSAVER_ACTIVE   717
diff -Naurp kodi-16.0-beta1-9a5ddf5/xbmc/GUIInfoManager.cpp kodi-16.0-beta1-9a5ddf5-patch/xbmc/GUIInfoManager.cpp
--- kodi-16.0-beta1-9a5ddf5/xbmc/GUIInfoManager.cpp	2015-11-15 19:10:48.000000000 +0200
+++ kodi-16.0-beta1-9a5ddf5-patch/xbmc/GUIInfoManager.cpp	2015-11-15 19:20:09.000000000 +0200
@@ -296,7 +296,7 @@ const infomap system_labels[] =  {{ "has
                                   { "profilecount",     SYSTEM_PROFILECOUNT },
                                   { "profileautologin", SYSTEM_PROFILEAUTOLOGIN },
                                   { "progressbar",      SYSTEM_PROGRESS_BAR },
-                                  { "batterylevel",     SYSTEM_BATTERY_LEVEL },
+                                  { "linuxver",         SYSTEM_LINUX_VER },
                                   { "friendlyname",     SYSTEM_FRIENDLY_NAME },
                                   { "alarmpos",         SYSTEM_ALARM_POS },
                                   { "isinhibit",        SYSTEM_ISINHIBIT },
@@ -1897,7 +1897,7 @@ std::string CGUIInfoManager::GetLabel(in
   case SYSTEM_INTERNET_STATE:
   case SYSTEM_UPTIME:
   case SYSTEM_TOTALUPTIME:
-  case SYSTEM_BATTERY_LEVEL:
+  case SYSTEM_LINUX_VER:
     return g_sysinfo.GetInfo(info);
     break;
 
@@ -2358,9 +2358,6 @@ bool CGUIInfoManager::GetInt(int &value,
     case PVR_TIMESHIFT_PROGRESS:
       value = g_PVRManager.TranslateIntInfo(info);
       return true;
-    case SYSTEM_BATTERY_LEVEL:
-      value = g_powerManager.BatteryLevel();
-      return true;
   }
   return false;
 }
diff -Naurp kodi-16.0-beta1-9a5ddf5/xbmc/utils/SystemInfo.cpp kodi-16.0-beta1-9a5ddf5-patch/xbmc/utils/SystemInfo.cpp
--- kodi-16.0-beta1-9a5ddf5/xbmc/utils/SystemInfo.cpp	2015-11-15 19:10:48.000000000 +0200
+++ kodi-16.0-beta1-9a5ddf5-patch/xbmc/utils/SystemInfo.cpp	2015-11-15 19:30:55.000000000 +0200
@@ -249,9 +249,9 @@ bool CSysInfoJob::DoWork()
   m_info.systemTotalUptime = GetSystemUpTime(true);
   m_info.videoEncoder      = GetVideoEncoder();
   m_info.cpuFrequency      = GetCPUFreqInfo();
-  m_info.osVersionInfo     = CSysInfo::GetOsPrettyNameWithVersion() + " (kernel: " + CSysInfo::GetKernelName() + " " + CSysInfo::GetKernelVersionFull() + ")";
+  m_info.osVersionInfo     = CSysInfo::GetOsPrettyNameWithVersion();
   m_info.macAddress        = GetMACAddress();
-  m_info.batteryLevel      = GetBatteryLevel();
+  m_info.linuxver          = CSysInfo::GetKernelVersionFull();
   return true;
 }
 
@@ -290,11 +290,6 @@ std::string CSysInfoJob::GetVideoEncoder
   return "GPU: " + g_Windowing.GetRenderRenderer();
 }
 
-std::string CSysInfoJob::GetBatteryLevel()
-{
-  return StringUtils::Format("%d%%", g_powerManager.BatteryLevel());
-}
-
 double CSysInfoJob::GetCPUFrequency()
 {
 #if defined (TARGET_POSIX) || defined(TARGET_WINDOWS)
@@ -380,8 +375,8 @@ std::string CSysInfo::TranslateInfo(int
       return g_localizeStrings.Get(13296);
     else
       return g_localizeStrings.Get(13297);
-  case SYSTEM_BATTERY_LEVEL:
-    return m_info.batteryLevel;
+  case SYSTEM_LINUX_VER:
+    return m_info.linuxver;
   default:
     return "";
   }
@@ -567,18 +562,20 @@ std::string CSysInfo::GetKernelName(bool
 std::string CSysInfo::GetKernelVersionFull(void)
 {
   static std::string kernelVersionFull;
+  static std::string kernelVersionR;
+  static std::string kernelVersionV;
+  static std::string kernelVersionM;
   if (!kernelVersionFull.empty())
     return kernelVersionFull;
 
-#if defined(TARGET_WINDOWS)
-  OSVERSIONINFOEXW osvi;
-  if (sysGetVersionExWByRef(osvi))
-    kernelVersionFull = StringUtils::Format("%d.%d", osvi.dwMajorVersion, osvi.dwMinorVersion);
-#elif defined(TARGET_POSIX)
   struct utsname un;
   if (uname(&un) == 0)
-    kernelVersionFull.assign(un.release);
-#endif // defined(TARGET_POSIX)
+  {
+    kernelVersionR.assign(un.release);
+    kernelVersionV.assign(un.version);
+    kernelVersionM.assign(un.machine);
+  }
+  kernelVersionFull = kernelVersionR + " " + kernelVersionV + " " + kernelVersionM;
 
   if (kernelVersionFull.empty())
     kernelVersionFull = "0.0.0"; // can't detect
diff -Naurp kodi-16.0-beta1-9a5ddf5/xbmc/utils/SystemInfo.h kodi-16.0-beta1-9a5ddf5-patch/xbmc/utils/SystemInfo.h
--- kodi-16.0-beta1-9a5ddf5/xbmc/utils/SystemInfo.h	2015-11-02 11:54:40.000000000 +0200
+++ kodi-16.0-beta1-9a5ddf5-patch/xbmc/utils/SystemInfo.h	2015-11-15 19:32:08.000000000 +0200
@@ -53,7 +53,7 @@ public:
   std::string cpuFrequency;
   std::string osVersionInfo;
   std::string macAddress;
-  std::string batteryLevel;
+  std::string linuxver;
 };
 
 class CSysInfoJob : public CJob
@@ -72,7 +72,6 @@ private:
   static std::string GetCPUFreqInfo();
   static std::string GetMACAddress();
   static std::string GetVideoEncoder();
-  static std::string GetBatteryLevel();
 
   CSysData m_info;
 };
diff -Naurp kodi-16.0-beta1-9a5ddf5/xbmc/windows/GUIWindowSystemInfo.cpp kodi-16.0-beta1-9a5ddf5-patch/xbmc/windows/GUIWindowSystemInfo.cpp
--- kodi-16.0-beta1-9a5ddf5/xbmc/windows/GUIWindowSystemInfo.cpp	2015-11-15 19:10:48.000000000 +0200
+++ kodi-16.0-beta1-9a5ddf5-patch/xbmc/windows/GUIWindowSystemInfo.cpp	2015-11-15 19:33:17.000000000 +0200
@@ -100,7 +100,7 @@ void CGUIWindowSystemInfo::FrameMove()
     SetControlLabel(i++, "%s %s", 13283, SYSTEM_OS_VERSION_INFO);
     SetControlLabel(i++, "%s: %s", 12390, SYSTEM_UPTIME);
     SetControlLabel(i++, "%s: %s", 12394, SYSTEM_TOTALUPTIME);
-    SetControlLabel(i++, "%s: %s", 12395, SYSTEM_BATTERY_LEVEL);
+    SetControlLabel(i++, "%s: %s", 12395, SYSTEM_LINUX_VER);
   }
 
   else if (m_section == CONTROL_BT_STORAGE)
